### PR TITLE
Feat: Add Markdown table report for benchmark results

### PR DESCRIPTION
**Pull request by presham**
## Summary
- Add new benchmark formatting logic (`formatDuration`, `formatBytes`, `toMarkdown`) to create human-readable report `report.md` to an existing `benchmark/write-snapshot.ts` module
- Extract benchmark formatting logic (`formatDuration`, `formatBytes`, `toMarkdown`) into a new `benchmark/markdown.ts` module
- Update `write-snapshot.ts` to import from `markdown.ts` and write a human-readable `report.md` alongside the existing `report.json`
- Re-run benchmark snapshot (`report.json` + new `report.md`)
- Add unit tests for all three exported functions covering unit-boundary conversions, the `—` fallback when heap/gc data is absent, and multi-row rendering

## Linked Issue
Closes: https://github.com/NodeSecure/js-x-ray/issues/612

## Verification
- [x] `node --test workspaces/js-x-ray/test/markdown.spec.ts` passes
      (14 tests across `formatDuration`, `formatBytes`, `toMarkdown`)
- [x] Existing test suite has no regressions
- [x] `npx eslint workspaces` reports no errors
- [x]  Build pass `npm run build`

### Report
image

### Test
image

### CODE REVIEW

**Comment by presham** [comment]
Hi @clemgbld 
This is my first contribution to this project. I've addressed [#612] by adding a readable format for the benchmark results. Would appreciate a review when you have time!

File: workspaces/js-x-ray/benchmark/write-snapshot.ts [hunk 1]
```diff
@@ -30,3 +33,6 @@ const relevantResults = {
 };
 
 writeFileSync(kReportURL, JSON.stringify(relevantResults, null, 2));
```
Comment (clemgbld) [inline workspaces/js-x-ray/benchmark/write-snapshot.ts]: In my opinion we can stop writing in the report.json file now that since we have a prettier result, you can also remove the report.json file i guess

**Comment by clemgbld** [comment]
@resham57 Thanks for your contribution ! can you add the changeset 
https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md

File: workspaces/js-x-ray/test/markdown.spec.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    assert.strictEqual(formatBytes(5_242_880), "5.00 MB");
+  });
+
+  test("should format values at or above 1,073,741,824 as GB", () => {
+    assert.strictEqual(formatBytes(1_073_741_824), "1.00 GB");
+    assert.strictEqual(formatBytes(2_684_354_560), "2.50 GB");
+  });
+});
+
+describe("toMarkdown", () => {
```
Comment (clemgbld) [inline workspaces/js-x-ray/test/markdown.spec.ts:8]: We can keep the test of toMarkdown and delete the others i guess since it is the only function used in write-snapshot.ts, you can also export only toMarkdown in the production code

File: workspaces/js-x-ray/benchmark/write-snapshot.ts (hunk 1 above)
Comment (presham) [reply workspaces/js-x-ray/benchmark/write-snapshot.ts]: Both report.json and its generation code have been removed.

File: workspaces/js-x-ray/test/markdown.spec.ts (hunk 2 above)
Comment (presham) [reply workspaces/js-x-ray/test/markdown.spec.ts:8]: Removed `formatDuration` and `formatBytes` specs, also removed export from `formatDuration` and `formatBytes` helpers

**Comment by presham** [comment]
> @resham57 Thanks for your contribution ! can you add the changeset https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md

It's my pleasure, @clemgbld. I've applied the changes you have mentioned and added a new changeset. Could you please review them once more and let me know if any further changes changes are required? I'll be happy to make them. Thank you!

**Review by clemgbld — APPROVED** [review]

**Review by fraxken — APPROVED** [review]

**Comment by fraxken** [comment]
@allcontributors please add @presham for code, benchmark
